authors/apps/profiles/views.py

```diff
@@ -61,3 +63,64 @@ class UserProfileListView(generics.ListAPIView):
     @classmethod
     def get_queryset(self):
         return Profile.objects.all()
+
+
+class IFollowListView(generics.RetrieveAPIView):
+    """A class for following and or unfollowing other authors."""
+    permission_classes = (IsAuthenticated,)
+    serializer_class = FollowerSerializer
+
+    @classmethod
+    def get(cls, request):
+        # returns a list of actors who request.user is following
+        data = {'id': request.user.pk}
+        following = Follower.objects.values_list('author', flat=True).filter(follow=request.user.username)
+        response = {'message': f'You follow : {following}', 'status_code': status.HTTP_200_OK}
+        return Response(response, content_type='text/json', status=response['status_code'])
+
+
+class FollowsUpdateAPIView(generics.UpdateAPIView):
+    """A class for following and or unfollowing other authors."""
+    permission_classes = (IsAuthenticated,)
+    lookup_field = 'username'
+    serializer_class = FollowerSerializer
+
+    @classmethod
+    def put(cls, request, username):
+        # Follow the author (where an author username=username).
+        data = {'username': request.user.username}
+        user = get_object_or_404(User, username=username)
+        try:
+            already = get_object_or_404(Follower, author=user.username, follow=data['username'])
+            response = {'error': f'You are already a follower to : {user.username}',
+                        'status_code': status.HTTP_400_BAD_REQUEST}
+        except Exception as e:
+            follow = Follower.objects.create(author=user.username, follow=data['username'])
+            follow.save()
+            response = {'massage': f'You are now a follower to : {user.username}', 'status_code': status.HTTP_200_OK}
+        return Response(response, content_type='text/json', status=response['status_code'])
+
+    @classmethod
+    def patch(cls, request, username):
+        # Stop following the author (where an author username=username).
+        data = {'username': request.user.username}
+        try:
+            unfollow = get_object_or_404(Follower, author=username, follow=data['username'])
+            unfollow.delete()
+            response = {'massage': f'You are nolonger a follower to : {username}', 'status_code': status.HTTP_200_OK}
+        except Exception as e:
+            response = {'error': f'You are not a follower to : {username}', 'status_code': status.HTTP_400_BAD_REQUEST}
```

Comment:
@ManuelDominic, use the readme for what should be returned https://github.com/mirest/ah-haven-space-sprinters#follow-user